src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -125,65 +135,140 @@ const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelec
     setOpenKeys({ ...openKeys, [item.path]: !openKeys[item.path] });
   };
 
+  const handleDownArrow = () => {
+    if (cursor.current + 1 < visibleNodes.current.length) {
+      cursor.current += 1;
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    }
+  };
+
+  const handleUpArrow = () => {
+    if (cursor.current >= 1) {
+      cursor.current -= 1;
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    }
+  };
+
+  const findParent = () => {
+    if (!currentNodeId) return;
+    const parentId = visibleNodes.current.find((el) => el.id === currentNodeId).parent;
+    if (parentId !== '') {
+      cursor.current = visibleNodes.current.findIndex((el) => el.id === parentId);
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    }
+  };
+
+  const findNode = (char) => {
+    let sortedNodes = visibleNodes.current.slice(cursor.current + 1, visibleNodes.current.length);
+    sortedNodes = sortedNodes.concat(visibleNodes.current.slice(0, cursor.current));
+    const match = sortedNodes.find((el) => el.id[0].toUpperCase() === char);
+    if (match) {
+      cursor.current = visibleNodes.current.findIndex((el) => el.id === match.id);
+      setCurrentNodeId(match.id);
+    }
+  };
+
   const handleKeyDown = (event, item) => {
     if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
       event.preventDefault();
       handleOnClick(event, item);
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_DOWN) {
```

Comment:
This function got a lot more dense with all of your enhancements in here. Can we break this up somewhat? Make dedicated functions for the different keys? Use a switch/case instead of the cascading ifs? I'd prioritize readability over minimizing a little duplication, if it comes down to that.

Reply:
I just changed it to a switch statement, but it could still be a bit cleaner I think. Let me know what you think.